[Enhancement] Start timer only after initial button press and display seconds left and average result

Tauka opened the pull request:
* I think this will make game experience richer, cause now you would have time to mentally prepare yourself before start (honestly, I expected this behavior by default).
* Also I would like to know my average score, so If I fail before timer expires, I would know what score I would ideally reach

ruskakimov commented:
also you can deploy your changes by

1. installing surge first (`npm i -g surge`)
2. changing the domain name in deploy script (`package.json`)
3. running `npm run deploy` or `yarn run deploy`

Tauka commented:
#### I made requested changes, please review them
> now `Expected score` displays `average * 60`, which is the score player would get if completed the game in the same pace

ruskakimov commented:
Could you also add current score to the overlay? :) It feels like it's missing. After I will merge and deploy it.

Tauka commented:
* Added `Your Score` to overlay
* Added `Go to Main` button
> This was done in order to avoid accidentally clicking away overlay during intensive gameplay

ruskakimov commented on src/components/GameScreen/index.js:
```diff
@@ -54,10 +54,23 @@ const OverlayChild = styled.div`
+  font-weight: bold;
+  cursor: pointer;
+  /* color of button will be the same as the color of the last word */
+  background-color: ${({ color }) => color ? color : "#3764a9"};
```
is this condition necessary? `color` should always be set to something

Tauka replied:
Ideally yes, but it's a safety measure. As I am not aware of all effects of this application, I assumed that sometimes, for some reasons, color might get `undefined`. If you are sure it won't happen under any circumstances, I can remove condition

ruskakimov replied:
yep, remove it please. :) Other components use that value directly.

ruskakimov replied:
after that i can merge it

Tauka commented:
@ruskakimov Btw, how about adding keyboard support? So you can play it with left and right arrows, my phone does not support anything 😄

ruskakimov commented:
@Tauka you can do it if you want :) But better do it in a separate PR. I can even create an issue.

Tauka commented:
#### Changes has been made
